egress/src/reports/idf_station.rs

```diff
@@ -0,0 +1,238 @@
+use axum::{
+    extract::{Path, Query, State},
+    http::StatusCode,
+    Json,
+};
+use chrono::Utc;
+use serde::{Deserialize, Serialize};
+use util::PooledPgConn;
+
+use crate::{errors, PgConnectionPool};
+
+/// Unit of the intensity values in the response
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, Default)]
+#[serde(rename_all = "lowercase")]
+pub enum IdfUnit {
+    /// Millimeters
+    #[default]
+    Mm,
+    /// Litres per second per hectare
+    Lsha,
+}
+
+/// Precipitation intensity values fitted from a GEV distribution on annual precipitation timeseries.
+/// More information can be found [here](https://www.met.no/publikasjoner/met-report/met-report-2022) under the link titled "IVF-verdier for norske nedbørstasjoner".
+/// The code responsible for generating this values can be found [here](https://github.com/ClimDesign/fixIDF).
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct IdfValue {
+    /// Duration of the precipitation event in minutes
+    pub duration: i32,
+    /// Return period in years
+    pub frequency: i32,
+    /// Computed intensity value in millimeters (mm)
+    pub intensity: f64,
+    /// 0.025 quantile
+    pub lower_interval: f64,
+    /// 0.975 quantile
+    pub upper_interval: f64,
+}
+
+/// Metadata and parameters used for fitting IDF values
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct IdfMetadata {
+    #[serde(skip)]
+    tsid: i32,
+    number_of_seasons: i32,
+    // TODO: should we have these instead?
+    // fromtime: Option<DateTime<Utc>>,
+    // totime: Option<DateTime<Utc>>,
+    first_year_of_period: i32,
+    last_year_of_period: i32,
+    quality_class: i32,
+    seed_parameter: i32,
+    updated_at: chrono::DateTime<Utc>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct IdfStationInfo {
```

Comment:
This struct seems to be subset of `IdfMetadata`. It's unclear why they're separate? I assume there's a good reason, so we should clarify this with a comment and/or clearer naming and/or module structure

Reply:
True, this is only used for the 'availability' response. The intention was to mimic what ODA is doing, but since I changed that (point 1. in the first comment), I guess we can return all the metadata instead.

Reply:
Yes, thank you